app.spec.js

```diff
@@ -0,0 +1,105 @@
+import request from 'supertest';
+import '@babel/polyfill';
+import app from './app';
+
+describe('API', () => {
+  let notes;
+  beforeEach(() => {
+    notes = [
+      {title: 'Dylan', 
+      noteItems: [{text: 'Is great'}, {text: 'Also Cool'}], 
+      id: 1},
+      {title: 'Shannon', 
+      noteItems: [{text: 'Is amazing'}, {text: 'Also Beautiful'}], 
+      id: 2},
+      {title: 'Justin', 
+      noteItems: [{text: 'Is awesome'}, {text: 'Also stylish'}], 
+      id: 3}
+    ]
+    app.locals.notes = notes
+  })
+
+  describe('get /notes', () => {
+    it('should return status 200 and all of the notes', async () => {
+      //setup / execution
+      const response = await request(app).get('/notes');
+      //expectation
+      expect(response.status).toBe(200);
+      expect(response.body).toEqual(app.locals.notes);
+    });
+  });
+
+  describe('delete /notes/:id', () => {
+    it('should remove the note and send a 204 status code', async () => {
+      //setup
```

Comment:
The usage of set/execution/expectation makes it very clear what each section of code is for. ⭐️